src/hawkbit-client.c

```diff
@@ -249,7 +249,7 @@ static size_t curl_write_cb(void *content, size_t size, size_t nmemb, void *data
 static gint rest_request(enum HTTPMethod method, const gchar* url, JsonBuilder* jsonRequestBody, JsonParser** jsonResponseParser, GError** error)
 {
         gchar *postdata = NULL;
-        struct rest_payload fetch_buffer;
+        RestPayload *fetch_buffer = g_new0(RestPayload, 1);
```

Comment:
maybe note that you change from stack allocation to heap allocation here

Reply:
Done.